execnet/gateway_base.py

```diff
@@ -307,7 +329,7 @@ def spawn(self, func, *args, **kwargs):
             of the given func(*args, **kwargs).
         """
         reply = Reply((func, args, kwargs), self.execmodel)
-        with self._running_lock:
+        with self._semaphore, self._running_lock:
```

Comment:
This isn't right at all, it's just around the start and not waiting until the task is finished